lib/controllers/project.js

```diff
@@ -761,8 +760,31 @@ Project.prototype.detectProjectType = function (path, callback) {
   */
 //-----------------------------------------------------------------------------
 Project.prototype.isSupported = function (srcVersion, runtime) {
+  var imageVersion, majorPass, minorPass, patchPass;
```

Comment:
If this is risky, what's the best way to do this? Or is this goal not accomplished?